packages/client/src/createTRPCClientProxy.ts

```diff
@@ -48,15 +47,19 @@ type SubscriptionResolver<
 type DecorateProcedure<
   TProcedure extends Procedure<any>,
   TRouter extends AnyRouter,
-> = OmitNeverKeys<{
-  query: TProcedure extends { _query: true } ? Resolver<TProcedure> : never;
-
-  mutate: TProcedure extends { _mutation: true } ? Resolver<TProcedure> : never;
-
-  subscribe: TProcedure extends { _subscription: true }
-    ? SubscriptionResolver<TProcedure, TRouter>
-    : never;
-}>;
+> = TProcedure extends { _type: 'query' }
```

Comment:
Would `QueryProcedure` have the same perf gain? If so, it might be better for us to use that for readability, I never liked how we were using implicit types that aren't explicitly "owned" by the `@trpc/server` package

Reply:
If not, we could probably do a `QueryProcedureTag` on the server at least

Reply:
I don't know. I have no intuition on how it would affect performance.